### PR TITLE
Use dhcp.client link attribute to enable DHCP on all non-relay nodes

### PR DESCRIPTION
**Pull request by ipspace**
Addresses the configuration part of #1343. The DHCP pools for the VLAN segments are not correct (duplicate pools, missing 'excluded' list)

### CODE REVIEW

**Comment by ipspace** [comment]
See #1358 for more details on the DHCP pools issue

File: docs/module/dhcp.md [hunk 1]
```diff
@@ -56,17 +56,21 @@ DHCP relays are supported on these platforms:
...
+
+You can enable the DHCP client on all non-relaying devices attached to a link with the boolean **dhcp.client.ipv4** link attribute and the DHCPv6 client with the boolean **dhcp.client.ipv6** attribute.
+
 ## Interface Parameters
 
-* To start a DHCP client on an interface, set **ipv4** or **ipv6** address to **dhcp**.
-* The DHCP client usually installs a default route to an IPv4 default gateway[^DGv6] sent in a DHCP reply. To disable this functionality, set **dhcp.client.default** interface parameter to `False`.
-* To start a DHCP relay on an interface, set **dhcp.server** interface parameter to a list of DHCP servers.
+* To start a DHCP client on an interface, set the **ipv4** or **ipv6** address to **dhcp**. You can use the **dhcp.client.ipv4** or the **dhcp.client.ipv6** interface attribute only when the node already uses the DHCP module. 
+* The DHCP client usually installs a default route to an IPv4 default gateway[^DGv6] sent in a DHCP reply. To turn off this functionality, set **dhcp.client.default** interface parameter to `False`.
```
Comment (jbemmel) [inline docs/module/dhcp.md:72]: Shouldn't this be ```dhcp.client.ipv4.default```?

File: netsim/modules/dhcp.py [hunk 2]
```diff
@@ -35,11 +35,14 @@ def check_protocol_support(node: Box, topology: Box) -> bool:
   # DHCP clients
   #
   for intf in node.interfaces:
-    if not intf.get('dhcp.client',False):
+    if not intf.get('dhcp.client',False):         # Not a DHCP client? OK
       continue
+    if intf.get('dhcp.server',False):             # Also a DHCP relay? That can't be
+      intf.dhcp.pop('client',None)                # ... remove the client flag
```
Comment (jbemmel) [inline netsim/modules/dhcp.py:41]: Log a warning?

File: netsim/modules/dhcp.py [hunk 3]
```diff
@@ -35,11 +35,14 @@ def check_protocol_support(node: Box, topology: Box) -> bool:
...
+      continue                                    # ... and move on
     for af in ('ipv4','ipv6'):
       if not af in intf.dhcp.client:
-        continue
+        continue
```
Comment (jbemmel) [inline netsim/modules/dhcp.py:45]: white space not needed

**Review by jbemmel — COMMENTED** [review]
Some minor comments, no show stoppers

File: docs/module/dhcp.md (hunk 1 above)
Comment (ipspace) [reply docs/module/dhcp.md:72]: If we set **dhcp.client.ipv4** to **True**, how can we then set **dhcp.client.ipv4.default** to something?

Comment (jbemmel) [reply docs/module/dhcp.md:72]: Various options:
1. dhcp.client.ipv4:
      request: True
      default: False
2. dhcp.client.ipv4_default: False

It's just to emphasize that this applies to ipv4 only

Comment (ipspace) [reply docs/module/dhcp.md:72]: DHCPv6 does not have a "default gateway" option. IPv6 uses RA to get the default gateway information.

**Pull request merged**